src/python/grpcio/tests/unit/framework/common/test_control.py

```diff
@@ -72,19 +73,31 @@ def control(self):
       if self._fail:
         raise Defect()
 
-      while self._paused:
+      while self._pause:
+        self._paused = True
+        self._condition.notify_all()
         self._condition.wait()
+      self._paused = False
 
   @contextlib.contextmanager
   def pause(self):
     """Pauses code under control while controlling code is in context."""
     with self._condition:
-      self._paused = True
+      self._pause = True
     yield
     with self._condition:
```

Comment:
Would it make sense to make ```paused``` be private, and call it after the ```yield```?  Or are there possible cases where the control never gets paused but we want to unpause it?